### DISCUSSION TITLE
Can't use custom 3D model for visualization

### DISCUSSION THREAD

**Discussion by AgentCosmic** [OpenModelica/OpenModelica#9547] [Q&A] [answered]
I'm using the `Modelica.Mechanics.MultiBody.Visualizers.FixedShape` to render custom 3D models. I tried using `.dxf` and `.stl` in ASCII format but none works.
For `.dxf` I get no error but the model doesn't show up.
For `.stl` I get an error saying there is no plugin for that file type.

**Comment by casella** [comment]
@adeas31 can you please help with that?

**Comment by adeas31** [comment]
This should be fixed with 6814223cf8a279f79d8f048b73edc96282e4f4ee. I haven't tested the `stl` but `dxf` should work.

**Comment by anotheruserofgithub** [comment]
To load the STL file, OSG's `ReaderWriterSTL` plugin is used: https://github.com/openscenegraph/OpenSceneGraph/blob/OpenSceneGraph-3.2/src/osgPlugins/stl/ReaderWriterSTL.cpp#L344

Could you tell the OpenSceneGraph version you are using?

**Comment by anotheruserofgithub** [comment]
I'm linking an answer to your original SO post about correct file path because it might be very useful for other users:
https://stackoverflow.com/a/74149149

**Comment by anotheruserofgithub** [comment]
@AgentCosmic
Could you please try to simulate [the attached `stl` model](https://github.com/OpenModelica/OpenModelica/files/10664594/stl.zip)? It should display a red dice. If not, please report the errors.

**Comment by anotheruserofgithub** [comment, marked as answer]
@AgentCosmic This issue has been solved. Thank you for reporting it! The fix is already available in the [nightly builds](https://build.openmodelica.org/omc/builds/windows/nightly-builds/) and will be shipped in the forthcoming [release 1.21](https://github.com/OpenModelica/OpenModelica/milestone/41).
